### PR TITLE
Make Jupyter Notebook 4.x the minimum baseline

### PR DESCRIPTION
**Pull request by parente**
Fixes issue #49

Merging pending ability to test and update declarative widget dashboard examples based on jupyter-incubator/declarativewidgets#67

### CODE REVIEW

**Comment by parente** [comment]
@lbustelo Have a look at this PR for all the Makefile, etc. updates. Widgets are going to differ of course, but it's a starting point. I'll hold this PR here until widgets have a 4.x compatibility PR that I can finish testing all our widget dashboard examples against.

**Comment by parente** [comment]
Rebased today since master has change quite a bit since I opened the PR.

File: Makefile [hunk 1]
```diff
@@ -1,74 +1,68 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-.PHONY: build clean dev help sdist test install remote
+.PHONY: clean dev dev-with-widgets help install js sdist test 
+
+REPO:=jupyter/pyspark-notebook:a388c4a66fd4
```
Comment (wellecks) [inline Makefile]: Why the use of this tag vs jupyter/pyspark-notebook:4.0 ?

Comment (parente) [inline Makefile]: https://github.com/jupyter/docker-stacks#stacks-tags-versioning-and-progress

tl;dr: 4.0 is legacy and no longer moving.

**Comment by parente** [comment]
Rebased on master (0.2.0.dev). Added two additional commits to account for changes. Manually tested all demos that do not require declarative widgets.

**Comment by parente** [comment]
Tested a whole slew of demos using what's in the jupyter-incubator/declarativewidgets#114 PR. Fixed on e error in the dashboard.js (init function now required). Everything else was solid.
